src/mochaVErrorReporter.ts

```diff
@@ -0,0 +1,46 @@
+import { Runner, utils as MochaUtils } from "mocha";
+import { VError } from "verror";
+
+var stackFilter = MochaUtils.stackTraceFilter();
+
+export class MochaVErrorReporter extends Mocha.reporters.Base
+{
+    private static _DEFAULT_INDENT: string = "    ";
+    private static _REGEX_REMOVE_REPEATED_MESSAGE: RegExp = /caused by.+\n/;
+    private static _REGEX_REPLACE_WITH_INDENT: RegExp = /\n/g;
+
+    constructor(runner: Runner) {
+        super(runner);
+        runner.on('fail', function(test, err) {
+            err.stack = stackFilter(MochaVErrorReporter.fullStack(err));
+          });
+    }
+
+    public static fullStack(error, indent = MochaVErrorReporter._DEFAULT_INDENT, indentLevel = 1)
```

Comment:
make this private ``fullStackImpl`` and have the public one without ``indentationLevel``  (rename `indentLevel`)

Reply:
I thought it would be a good idea to have an option to start indentation from a certain level. Looking from this standpoint, it's not only for internal use

Reply:
Refactored the parameter name

Reply:
Fixed after conversation